nhashing/nhashing.py

```diff
@@ -0,0 +1,167 @@
+import pandas as pd
+import multiprocessing
+from category_encoders.hashing import HashingEncoder
+
+__author__ = 'LiuShulun'
+
+
+class NHashingEncoder:
+    """A extended multivariate hashing implementation using multi-process
+
+    !!! Significant !!! improve the encoding speed of hashing encoder on
+    multi-kernel device. This NHashingEncoder allows setting
+    max_process & max_sample to use more CPU resource when hashing encoding.
+    On a 4C8T CPU, NHashingEncoder with max_process=4 makes
+    encoding 3+ times faster than hashingEncoder.
+
+    Parameters
+    ----------
+
+    verbose: int
+        integer indicating verbosity of the output. 0 for none.
+    cols: list
+        a list of columns to encode, if None all string columns'll be encoded.
+    drop_invariant: bool
+        boolean for whether or not to drop columns with 0 variance.
+    return_df: bool
+        boolean for whether to return a pandas DataFrame from transform
+        (otherwise it will be a numpy array).
+    hash_method: str
+        which hashing method to use. Any method from hashlib works.
+    max_process: int
+        how many PROCESS to use, NOT Thread. limited in range(1, 64)
+    max_sample: int
+        how many samples will be encode by each process at each time
+
+    Example
+    -------
+    >>> from category_encoders.hashing import HashingEncoder
+    >>> from n_hashing import NHashingEncoder
+    >>> import time
+    >>> import pandas as pd
+    >>> from sklearn.datasets import load_boston
+    >>> bunch = load_boston()
+    >>> X = pd.DataFrame(bunch.data, columns=bunch.feature_names)
+    >>> DL = []
+    >>> for i in range(1000): DL.append(X)
+    >>> DF = pd.concat(DL, ignore_index=True).reset_index(drop=True)
+    >>> he = HashingEncoder(cols=['CHAS', 'RAD'])
+    >>> start = time.time()
+    >>> he.fit_transform(DF)
+    >>> he_time = time.time() - start
+    >>> nhe = NHashingEncoder(cols=['CHAS', 'RAD'])
+    >>> start = time.time()
+    >>> nhe.fit_transform(DF)
+    >>> nhe_time = time.time() - start
+    >>> print("500000samples HashingEncoder Time:", he_time, "NHashingEncoder Time:", nhe_time)
```

Comment:
I suspect the lint process is just complaining about these lines over 79 characters

Reply:
Thanks for reading this! Sorry I don't know much about joblib and what it can do except saving model, but thanks for the suggestion! But openmp seems working just like multiprocessing, so...I do have considered multithreading, but it's thread after all, seems it can only create thread inside the process, and still waste other cores of CPU?

Reply:
I asked a guy about openmp, sadly I was totally wrong. Openmp's faster than multiprocessing, but..cython's unfamiliar to me.
Maybe there're some api based on cpython that support multi-process except multiprocessing? Are they faster than it?